morph_tool/nrnhines.py

```diff
@@ -273,3 +278,91 @@ def point_to_section_end(sections: Sequence[neuron.nrn.Section],
         if np.isclose(point, last_section_point, atol=atol, rtol=rtol).all():
             return index
     return None
+
+
+class NoDaemonProcess(multiprocessing.Process):
+    """Class that represents a non-daemon process"""
+
+    def _get_daemon(self):  # pylint: disable=R0201
+        """Get daemon flag"""
+        return False
+
+    def _set_daemon(self, value):
+        """Set daemon flag"""
+
+    daemon = property(_get_daemon, _set_daemon)
+
+
+class NestedPool(multiprocessing.pool.Pool):  # pylint: disable=abstract-method
+    """Class that represents a MultiProcessing nested pool"""
+
+    Process = NoDaemonProcess
+
+
+def isolate(func):
+    """Isolate a generic function for independent NEURON instances.
+
+    Note: it does not work as decorator.
+
+    Args:
+        func (function): function to isolate"""
+    def func_isolated(*args, **kwargs):
+        with NestedPool(1, maxtasksperchild=1) as pool:
+            return pool.apply(func, args, kwargs)
+    return func_isolated
+
+
+def convert_point_to_isec(morphology_path, point, section_type="apical"):
+    """Convert a point position to NEURON section index and return cell name and id.
+
+    Args:
+        morphology_path (Path): path to morphology
+        point (list/ndarray): 3d point
+    """
+    cell = get_NRN_cell(morphology_path)
+    point = point_to_section_end(getattr(cell.icell, section_type), point)
+    return morphology_path.stem, point
+
+
+def convert_point_to_isec_isolated(morph_data, section_type="apical"):
+    """Serves as a decorator to isolate convert_point_to_isc, but work with multiprocessing."""
+    return isolate(convert_point_to_isec)(*morph_data, section_type=section_type)
+
+
+def convert_points_to_isecs(morphology_path,
+                            points_file, isections_file,
+                            section_type='all', n_workers=1):
+    """Convert 3d positions to NEURON isections and save to json/yaml file.
```

Comment:
It's better to have functions that takes and return pure python objects. Maybe the user has the point list coming out from the result of a computation and not from a file. Maybe he does not want to write the output to disk but reuse it on the fly.

But once you have removed the I/O of this function, it only remains the for loop on the isolated function. So not so much.
However I'll try to improve the magic to isolate NEURON.

Reply:
yes, I agree. So the IO would rather go to an app? But we don't really want an app. I'm happy to just discard the IO from morph-tool and let is up be to the user.